src/auth/IdBroker.php

```diff
@@ -45,10 +47,28 @@ public function getAuthenticatedUser(string $username, string $password)
             'password' => $password,
         ]);
         $statusCode = $result['statusCode'] ?? null;
-        if (intval($statusCode) === 200) {
-            unset($result['statusCode']);
-            return $result;
+        switch (intval($statusCode)) {
+            
+            case 200: // Credentials were acceptable.
+                unset($result['statusCode']);
+                /*
+                 * Make sure all values are arrays themselves for simplesamlphp compatibility
+                 */
+                foreach ($result as $key => $value) {
+                    if ( ! is_array($value)) {
+                        $result[$key] = [$value];
+                    }
+                }
+                return $result;
+            
+            case 400: // Credentials were NOT acceptable.
+                return null;
+
+            default:
+                throw new \Exception(sprintf(
+                    'Unexpected response from ID Broker: %s',
+                    var_export($result->toArray(), true)
```

Comment:
Update: I'm correcting a bug in the ID Broker PHP Client so that the values returned will actually be arrays (not `GuzzleHttp\Command\Result` objects). We can then simply remove `->toArray()` here and be good to go.